File: Python/BOJ/7576.py
import sys
readl = sys.stdin.readline

dx = [-1, 1, 0, 0]; dy = [0, 0, -1, 1]

m, n = map(int, readl().split())
map = [list(readl().split()) for _ in range(n)]

total = n * m

queue = []
index = 0
visited = [[False for _ in range(m)] for _ in range(n)]
for i in range(n):
  for j in range(m):
    if map[i][j] == "1":
      queue.append((i, j, 0))
      visited[i][j] = True
    elif map[i][j] == "-1":
      total -= 1

result = 0
while index < len(queue):
  node = queue[index]
  index += 1

  result = max(result, node[2])
  
  for i in range(4):
    nx = node[0] + dx[i]
    ny = node[1] + dy[i]
    nc = node[2] + 1

    if not (0 <= nx < n and 0 <= ny < m):
      continue
    if not (not visited[nx][ny] and map[nx][ny] == "0"):
      continue

    visited[nx][ny] = True
    queue.append((nx, ny, nc))

if total > index:
  print("-1")
else:
  print(result)
  

File: Swift/BOJ/7000/7576.swift
//
//  7576.swift
//  Algorithm
//
//  Created by 유정주 on 2023/08/14.
//

import Foundation

typealias Point = (x: Int, y: Int, day: Int)

let dx = [-1, 1, 0, 0]
let dy = [0, 0, -1, 1]

let mn = readLine()!.split { $0 == " " }.map { Int(String($0))! }
let (m, n) = (mn[0], mn[1])

var queue: [Point] = []
var map: [[Int]] = []
var tomatoCount = 0
for i in 0..<n {
    let arr = readLine()!.split { $0 == " " }.map { Int(String($0))! }
    map.append(arr)
    
    for j in 0..<m {
        if map[i][j] == 1 {
            queue.append((i, j, 0))
        }
        
        if map[i][j] != -1 {
            tomatoCount += 1
        }
    }
}

var result = 0
var index = 0
while index < queue.count {
    let (x, y, day) = queue[index]
    index += 1
    
    for i in 0..<4 {
        let nx = x + dx[i]
        let ny = y + dy[i]
        
        guard 0..<n ~= nx, 0..<m ~= ny, map[nx][ny] == 0 else {
            continue
        }
        
        map[nx][ny] = 1
        result = max(result, day + 1)
        queue.append((nx, ny, day + 1))
    }
}

if tomatoCount == index {
    print(result)
} else {
    print("-1")
}
